Arzette/options.py

```diff
@@ -0,0 +1,171 @@
+    """Allows the Lichen Hills key to spawn locally in locations other than its vanilla location. This key is invisible until the Fatal Flute is played near it, making it difficult to find otherwise."""
+    internal_name = "shuffle_hills_key"
+    display_name = "Shuffle Hills Key"
+
+class ShuffleCandles(DefaultOnToggle):
+    """Include all candles and candle locations in the pool."""
+    internal_name = "shuffle_candles"
+    display_name = "Shuffle Candles"
+
+class ShuffleCoins(DefaultOnToggle):
+    """Include all coins and coin locations in the pool."""
+    internal_name = "shuffle_coins"
+    display_name = "Shuffle Coins"
+
+class ShufflePlants(DefaultOnToggle):
+    """Include all three of the Cypress' quest's plants (and their locations) in the pool."""
+    internal_name = "shuffle_plants"
+    display_name = "Shuffle Plants"
+
+class ShuffleUpgrades(DefaultOnToggle):
+    """Include all the capacity upgrades (and their locations) in the pool. This includes Infinite Soulfire."""
+    internal_name = "shuffle_upgrades"
+    display_name = "Shuffle Upgrades"
+
+class ShuffleLifeUps(DefaultOnToggle):
+    """Include all Life-Ups and Life-Up locations in the pool."""
+    internal_name = "shuffle_lifeups"
+    display_name = "Shuffle Life-Ups"
+
+class ShuffleBonusScrolls(Toggle):
+    """Include Bonus Scrolls in the item and location pools. Bonus Scrolls are always local, and always spawn in the open."""
+    internal_name = "shuffle_bonus_scrolls"
+    display_name = "Shuffle Bonus Scrolls"
+
+class ShuffleBonusScrollRewards(Toggle):
+    """Include the ruby rewards from the bonus minigame scrolls in the item and location pools."""
+    internal_name = "shuffle_bonus_rewards"
+    display_name = "Shuffle Bonus Scroll Rewards"
+
+class ShuffleRaceRewards(Toggle):
+    """Include the 100 ruby rewards from the Rudy races in the item and location pools."""
+    internal_name = "shuffle_race_rewards"
+    display_name = "Shuffle Race Rewards"
+
+class ShuffleBeacons(DefaultOnToggle):
+    """Include the Sacred Beacons in the item and location pools."""
+    internal_name = "shuffle_beacons"
+    display_name = "Shuffle Beacons"
+
+class ShuffleJewels(Toggle):
+    """Include the Faramore Jewel shards in the item and location pools. Beating the five bosses will now grant random items."""
+    internal_name = "shuffle_jewels"
+    display_name = "Shuffle Jewel Shards"
+
+class TradingSequence(Choice):
+    """
+Determine how the Chainsword trading sequence is randomized:
+Vanilla: All items except the Dungeon Key will be in their vanilla locations, including the Chainsword.
+Excluded: The Dungeon Key, the Sacred Oil, the Refined Chains, and the Chainsword will be shuffled. The rest of the trade quest will be avoided entirely, including Zazie's Soul Upgrade.
```

Comment:
```suggestion
Excluded: The Dungeon Key, the Sacred Oil & Refined Chains, and the Chainsword items will be shuffled, as well as the locations for the Dungeon Key, the Sacred Oil and the Chainsword. The rest of the trade quest will be avoided entirely, including Zazie's Soul Upgrade.
```

"Sacred Oil" and "Sacred Oil & Refined Chains" are two different items (and two different locations). The "excluded" option shuffles the "Sacred Oil" location and the "Sacred Oil & Refined Chains" item. It is the only option that includes a different item+location combo in the pool. This is what locks away the rest of the trading sequence.